internal/host/validations_test.go

```diff
@@ -2605,4 +2605,134 @@ var _ = Describe("Validations test", func() {
+		var (
+			cluster common.Cluster
+			host    models.Host
+		)
+		BeforeEach(func() {
+			cluster = hostutil.GenerateTestCluster(clusterID)
+			hostId, infraEnvId := strfmt.UUID(uuid.New().String()), strfmt.UUID(uuid.New().String())
+			mockProviderRegistry.EXPECT().IsHostSupported(commontesting.EqPlatformType(models.PlatformTypeVsphere), gomock.Any()).Return(false, nil).AnyTimes()
+			host = hostutil.GenerateTestHostByKind(hostId, infraEnvId, &clusterID, models.HostStatusKnown, models.HostKindHost, models.HostRoleMaster)
+			host.Inventory = hostutil.GenerateMasterInventory()
+			mNet := []*models.MachineNetwork{
+				{
+					Cidr:      "192.168.127.1/24",
+					ClusterID: clusterID,
+				},
+			}
+			cluster.MachineNetworks = mNet
+		})
+		It("Happy flow - MTU reports in the machine network are successful", func() {
+			connectivityReport := &models.ConnectivityReport{}
+			connectivityReport.RemoteHosts = append(connectivityReport.RemoteHosts, &models.ConnectivityRemoteHost{
+				HostID: hostID,
+				MtuReport: []*models.MtuReport{
+					{MtuSuccessful: true, RemoteIPAddress: "192.168.127.31", OutgoingNic: "ens3"},
+					{MtuSuccessful: false, RemoteIPAddress: "192.168.145.31", OutgoingNic: "ens4"},
+				},
+			})
+
+			report, err := hostutil.MarshalConnectivityReport(connectivityReport)
+			Expect(err).To(BeNil())
+			host.Connectivity = report
+			Expect(db.Create(&cluster).Error).ToNot(HaveOccurred())
+			Expect(db.Create(&host).Error).ShouldNot(HaveOccurred())
+
+			mockAndRefreshStatus(&host)
+
+			refreshedHost := hostutil.GetHostFromDB(*host.ID, host.InfraEnvID, db).Host
+
+			status, _, ok := getValidationResult(refreshedHost.ValidationsInfo, IsMtuOKInMachineNetwork)
+			Expect(ok).To(BeTrue(), fmt.Sprintf("debuuging info: status %s, isok: %t host: %#v cluster: %#v", status, ok, host, cluster))
+			Expect(status).To(Equal(ValidationSuccess))
+		})
+		It("Bad flow - MTU reports in the machine network are unsuccessful", func() {
+			connectivityReport := &models.ConnectivityReport{}
+			connectivityReport.RemoteHosts = append(connectivityReport.RemoteHosts, &models.ConnectivityRemoteHost{
+				HostID: hostID,
+				MtuReport: []*models.MtuReport{
+					{MtuSuccessful: false, RemoteIPAddress: "192.168.127.31", OutgoingNic: "ens3"},
+				},
+			})
+
+			report, err := hostutil.MarshalConnectivityReport(connectivityReport)
+			Expect(err).To(BeNil())
+			host.Connectivity = report
+			Expect(db.Create(&cluster).Error).ToNot(HaveOccurred())
+			Expect(db.Create(&host).Error).ShouldNot(HaveOccurred())
+
+			mockAndRefreshStatus(&host)
+
+			refreshedHost := hostutil.GetHostFromDB(*host.ID, host.InfraEnvID, db).Host
+
+			status, _, ok := getValidationResult(refreshedHost.ValidationsInfo, IsMtuOKInMachineNetwork)
+			Expect(ok).To(BeTrue(), fmt.Sprintf("debuuging info: status %s, isOK: %t host: %#v cluster: %#v", status, ok, host, cluster))
+			Expect(status).To(Equal(ValidationFailure))
+		})
+	})
```

Comment:
moving this to Table can shorten this code

Reply:
Agree. Thanks